src/Components/QuickSelect.tsx

```diff
@@ -1,202 +1,226 @@
-                  </Box>
-                ))}
-            </Box>
-            <Box className={props.classes.flexColumn}>
-              {props.recentlySelected
-                .slice(Math.floor(props.recentlySelected.length / 2))
-                .map((object) => (
-                  <Box>
-                    <Button
-                      onClick={() => handleClick(object)}
-                      color="primary"
-                      variant="text"
-                      size="small"
-                      disableFocusRipple={true}
-                      disableRipple={true}
-                      className={props.classes.quickSelectContainerButton}
-                    ></Button>
-                    <Box mt={-3}>
-                      <Typography color="primary" variant="subtitle2">
-                        {object.join(" ")}
-                      </Typography>
-                    </Box>
-                  </Box>
-                ))}
-            </Box>
-          </Box>
-        </Box>
-      </Box>
-    </Box>
-  );
-};
+import React, { useContext } from "react";
+import { Divider } from "@material-ui/core";
+import { Button, Box, Typography } from "@material-ui/core";
+import { DateSelect } from "./QuickDateSelect";
+import { DateRange, TermContext } from "./DateRange";
+import { GlobalContext } from "./Constants";
+import { DateRangeUI, DropdownData } from "./Types";
+import { MSFormatter } from "./MSFormatter";
+
+interface Inputs {
+  boxClass: string;
+  recentlySelected: number[];
+  dateRange: DateRange;
+  dateRangeUI: DateRangeUI;
+  dropdownData: DropdownData;
+  setDropdownData(data: DropdownData): void;
+  setDateRangeUI(dateRangeUI: DateRangeUI): void;
+  applyChanges(dr: DateRange): void;
+  setRecentlySelected(items: number[]): void;
+}
+
+export const QuickSelect: React.FC<Inputs> = (props) => {
+  const globals = useContext(GlobalContext);
+  let out = props.dropdownData.quickSelectContent;
+
+  let commonlyUsedFirstRow = globals.commonlyUsedText.slice(
+    0,
+    Math.floor(globals.commonlyUsedText.length / 2)
+  );
+
+  let commonlyUsedSecondRow = globals.commonlyUsedText.slice(
+    Math.floor(globals.commonlyUsedText.length / 2),
+    8
+  );
+
+  function getRecentlySelectedFirstRow(): number[] {
+    return props.recentlySelected.slice(
+      0,
+      Math.floor(props.recentlySelected.length / 2)
+    );
+  }
+
+  function getRecentlySelectedSecondRow(): number[] {
+    return props.recentlySelected.slice(
+      Math.floor(props.recentlySelected.length / 2)
+    );
+  }
+
+  function handleClick(text: number): void {
+    out = MSFormatter.splitMilliseconds(text);
+    props.setDropdownData({ ...props.dropdownData, quickSelectContent: out });
+  }
+
+  function apply(): void {
+    let dateRange = props.dateRange;
+    dateRange.setQuickSelect(props.dropdownData.quickSelectContent);
+    props.applyChanges(dateRange);
+
+    let recentlySelected = props.recentlySelected;
+    recentlySelected.unshift(
```

Comment:
Some comments might be worth adding here, indicating why this is being done.

Reply:
Done